client-course-schedulizer/src/components/reuseables/AddSectionPopover/AddSectionPopover.tsx

```diff
@@ -1,88 +1,252 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Button, Grid, Typography } from "@material-ui/core";
+import React, { ChangeEvent, useState } from "react";
+import { useForm } from "react-hook-form";
+import { array, object } from "yup";
 import {
-  Button,
-  Checkbox,
-  FormControl,
-  FormControlLabel,
-  FormLabel,
-  Grid,
-  Radio,
-  RadioGroup,
-  TextField,
-} from "@material-ui/core";
-import React from "react";
+  Course,
+  Day,
+  Half,
+  Intensive,
+  Section,
+  SemesterLength,
+  SemesterLengthOption,
+  Term,
+} from "../../../utilities/interfaces/dataInterfaces";
+import { GridItemCheckboxGroup } from "../GridItem/GridItemCheckboxGroup";
+import { GridItemRadioGroup } from "../GridItem/GridItemRadioGroup";
+import { GridItemTextField } from "../GridItem/GridItemTextField";
 import "./AddSectionPopover.scss";
 
+interface SectionInput {
```

Comment:
This is similar to the type `di.Section` right? Maybe that can be used? Maybe not and that's okay.

Reply:
I tried to do it with `di.Section` but I think that would mean the data from the form would need to be mapped using the yup resolver, which seemed a bit difficult. Maybe something to consider for the future though